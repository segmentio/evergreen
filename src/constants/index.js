export { default as StackingOrder } from './src/StackingOrder'
export { default as Intent } from './src/Intent'
export { default as Position } from './src/Position'
