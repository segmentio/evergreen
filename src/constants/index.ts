export { default as StackingOrder } from './src/StackingOrder'
export { default as Intent } from './src/Intent'
export { default as Position } from './src/Position'
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module './sr... Remove this comment to see the full error message
export { default as MimeType } from './src/MimeType'
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module './sr... Remove this comment to see the full error message
export { default as Key } from './src/Key'
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module './sr... Remove this comment to see the full error message
export { default as FileRejectionReason } from './src/FileRejectionReason'
