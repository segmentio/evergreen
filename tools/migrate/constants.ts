const INDEX_D_TS = 'index.d.ts'

export { INDEX_D_TS }
