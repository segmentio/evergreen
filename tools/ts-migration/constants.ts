const ICON_COMPONENT = 'IconComponent'
const INDEX_D_TS = 'index.d.ts'

export { ICON_COMPONENT, INDEX_D_TS }
