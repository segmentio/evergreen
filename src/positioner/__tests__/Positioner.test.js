import { Position } from '../../constants'
import getFittedPosition from '../src/getPosition'

const dimensions = overrides => ({
  height: 100,
  width: 100,
  ...overrides
})

const targetRect = overrides => ({
  x: 250,
  y: 150,
  width: 50,
  height: 30,
  top: 150,
  bottom: 150 - 30,
  left: 250,
  right: 200,
  ...overrides
})

const targetOffset = 6

const viewport = overrides => ({
  height: 250,
  width: 850,
  ...overrides
})

describe('<Positioner />', () => {
  it('All positions work', () => {
    const generatedPositions = Object.values(Position).map(position =>
      getFittedPosition({
        position,
        dimensions: dimensions(),
        targetRect: targetRect(),
        targetOffset,
        viewport: viewport()
      })
    )
    expect(generatedPositions).toMatchSnapshot()
  })

  it('Position.LEFT repositions to the right', () => {
    expect(
      getFittedPosition({
        position: Position.LEFT,
        dimensions: dimensions({ width: 350 }),
        targetRect: targetRect(),
        targetOffset,
        viewport: viewport()
      })
    )
  })

  it('Position.RIGHT repositions to the left', () => {
    expect(
      getFittedPosition({
        position: Position.RIGHT,
        dimensions: dimensions({ width: 250 }),
        targetRect: targetRect({ left: 800, x: 800, right: 850 }),
        targetOffset,
        viewport: viewport()
      })
    ).toMatchSnapshot()
  })

  it('Position.LEFT and Position.RIGHT will use the side with the most space', () => {
    expect(
      getFittedPosition({
        position: Position.LEFT,
        dimensions: dimensions({ width: 250 }),
        targetRect: targetRect({ left: 50, x: 50, right: 100 }),
        targetOffset,
        viewport: viewport({ width: 300 })
      })
    ).toMatchSnapshot()
  })

  it('Position.TOP repositions to the bottom', () => {
    expect(
      getFittedPosition({
        position: Position.TOP,
        dimensions: dimensions({ height: 250 }),
        targetRect: targetRect({ top: 20, y: 20 }),
        targetOffset,
        viewport: viewport()
      })
    ).toMatchSnapshot()
  })

  it('Position.BOTTOM repositions to the top', () => {
    expect(
      getFittedPosition({
        position: Position.BOTTOM,
        dimensions: dimensions({ height: 250 }),
        targetRect: targetRect({ top: 290, y: 290, bottom: 295, height: 5 }),
        targetOffset,
        viewport: viewport({ height: 300 })
      })
    ).toMatchSnapshot()
  })

  it('It pushes the rect to the right if overflowing on the left side', () => {
    expect(
      getFittedPosition({
        position: Position.BOTTOM,
        dimensions: dimensions({ width: 250, height: 110 }),
        targetRect: targetRect({ left: 10, x: 10, top: 10, y: 10, bottom: 20 }),
        targetOffset,
        viewport: viewport()
      })
    ).toMatchSnapshot()
  })

  it('It pushes the rect to the top if overflowing on the bottom side', () => {
    expect(
      getFittedPosition({
        position: Position.TOP,
        dimensions: dimensions({ width: 50, height: 100 }),
        targetRect: targetRect({ left: 100, x: 100, top: 550, y: 550 }),
        targetOffset,
        viewport: viewport({ height: 500 })
      })
    ).toMatchSnapshot()
  })
})
