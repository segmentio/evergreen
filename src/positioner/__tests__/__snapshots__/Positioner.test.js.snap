// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Positioner /> All positions work 1`] = `
Array [
  Object {
    "position": "top",
    "rect": Object {
      "bottom": 144,
      "height": 100,
      "left": 225,
      "right": 325,
      "top": 44,
      "width": 100,
    },
    "transformOrigin": "50px 100px ",
  },
  Object {
    "position": "top-left",
    "rect": Object {
      "bottom": 144,
      "height": 100,
      "left": 250,
      "right": 350,
      "top": 44,
      "width": 100,
    },
    "transformOrigin": "25px 100px ",
  },
  Object {
    "position": "top-right",
    "rect": Object {
      "bottom": 144,
      "height": 100,
      "left": 100,
      "right": 200,
      "top": 44,
      "width": 100,
    },
    "transformOrigin": "175px 100px ",
  },
  Object {
    "position": "bottom",
    "rect": Object {
      "bottom": 226,
      "height": 100,
      "left": 225,
      "right": 325,
      "top": 126,
      "width": 100,
    },
    "transformOrigin": "50px 0px ",
  },
  Object {
    "position": "bottom-left",
    "rect": Object {
      "bottom": 226,
      "height": 100,
      "left": 250,
      "right": 350,
      "top": 126,
      "width": 100,
    },
    "transformOrigin": "25px 0px ",
  },
  Object {
    "position": "bottom-right",
    "rect": Object {
      "bottom": 226,
      "height": 100,
      "left": 100,
      "right": 200,
      "top": 126,
      "width": 100,
    },
    "transformOrigin": "175px 0px ",
  },
  Object {
    "position": "left",
    "rect": Object {
      "bottom": 215,
      "height": 100,
      "left": 144,
      "right": 244,
      "top": 115,
      "width": 100,
    },
    "transformOrigin": "100px 50px",
  },
  Object {
    "position": "right",
    "rect": Object {
      "bottom": 215,
      "height": 100,
      "left": 206,
      "right": 306,
      "top": 115,
      "width": 100,
    },
    "transformOrigin": "0px 50px",
  },
]
`;

exports[`<Positioner /> It pushes the rect to the right if overflowing on the left side 1`] = `
Object {
  "position": "bottom",
  "rect": Object {
    "bottom": 136,
    "height": 110,
    "left": 8,
    "right": 258,
    "top": 26,
    "width": 250,
  },
  "transformOrigin": "27px 0px ",
}
`;

exports[`<Positioner /> It pushes the rect to the top if overflowing on the bottom side 1`] = `
Object {
  "position": "top",
  "rect": Object {
    "bottom": 492,
    "height": 100,
    "left": 100,
    "right": 150,
    "top": 392,
    "width": 50,
  },
  "transformOrigin": "25px 100px ",
}
`;

exports[`<Positioner /> Position.BOTTOM repositions to the top 1`] = `
Object {
  "position": "top",
  "rect": Object {
    "bottom": 284,
    "height": 250,
    "left": 225,
    "right": 325,
    "top": 34,
    "width": 100,
  },
  "transformOrigin": "50px 250px ",
}
`;

exports[`<Positioner /> Position.LEFT and Position.RIGHT will use the side with the most space 1`] = `
Object {
  "position": "right",
  "rect": Object {
    "bottom": 215,
    "height": 100,
    "left": 42,
    "right": 292,
    "top": 115,
    "width": 250,
  },
  "transformOrigin": "0px 50px",
}
`;

exports[`<Positioner /> Position.RIGHT repositions to the left 1`] = `
Object {
  "position": "left",
  "rect": Object {
    "bottom": 215,
    "height": 100,
    "left": 544,
    "right": 794,
    "top": 115,
    "width": 250,
  },
  "transformOrigin": "250px 50px",
}
`;

exports[`<Positioner /> Position.TOP repositions to the bottom 1`] = `
Object {
  "position": "bottom",
  "rect": Object {
    "bottom": 242,
    "height": 250,
    "left": 225,
    "right": 325,
    "top": -8,
    "width": 100,
  },
  "transformOrigin": "50px 0px ",
}
`;
